### DISCUSSION TITLE
Use public email for commits done with Web-based editor

### DISCUSSION THREAD

**Discussion by id-ilych** [community/community#6290] [Codespaces]
The web-based editor is awesome for review, researches, and small fixes 🚀 

Today I noticed though that the commits I did with this editor are signed with my personal email instead of the public email I've selected in the settings. Please add an option to configure it somehow (e.g. per project) because my work-related commits should be labeled with the business email.

Thank you for the feature!

**Comment by joyceerhl** [comment]
@id-ilych, just to confirm, did you configure your primary GitHub email as follows? https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/changing-your-primary-email-address We should certainly be using your primary GitHub email for all commits created through github.dev. I'm unfortunately unable to reproduce the behavior you're describing, and would greatly appreciate any extra details you can provide.

**Reply by id-ilych** [reply]
Hey @joyceerhl 👋 

Hmm. Now it's confusing. I don't have my business email as the primary (https://github.com/settings/emails) but I do have it as public one (https://github.com/settings/profile). ~~Then when I use "Squash and merge" feature on the PR then the committer is properly set to my business email. And thus I was expecting the same to happen for the commits done via github.dev 
So maybe it is not about "wrong" email being used by github.dev but rather about inconsistency with other behaviors~~
**Update** I double-checked and the "commiter" in that case is set to Github <noreply@github.com>.

Additional confusion comes from the fact that the public email vs primary email difference is not clear. What are their usages exactly then? Because for **primary email** it just says "will be used for account-related notifications and can be used for password resets." - there is nothing about commits at all. And for the **public email** there is no description at all.

So, can we have a **dedicated** configuration that is used for commits created via UI (both the main Github UI and the github.dev)? Ideally - a configuration that is set per organization. This way a lot of confusion could be avoided, and also people will not be forced to change their password-related email (as a bonus, it will also allow them to work in multiple organizations' repos correctly).

Thank you!

**Reply by joyceerhl** [reply]
AFAIK the public email controls which email is displayed on your profile page. For now I would recommend setting the primary GitHub email to ensure that your github.dev commits are created with the right email. Let us know if that doesn't behave as expected.

Good point on the consistency issues--will try to route that feedback to the right team.
